src/Lib.P2P.Tests/SwarmTest.cs

```diff
@@ -0,0 +1,1244 @@
+
+                await swarmA.StopListeningAsync(addr);
+                Assert.IsFalse(peerA.Addresses.Contains(another));
+            }
+            finally
+            {
+                await swarmA.StopAsync();
+                await swarmB.StopAsync();
+            }
+        }
+
+        [TestMethod]
+        public async Task Listening_IPv4Any()
+        {
+            var peerA = new Peer
+            {
+                Id = self.Id,
+                PublicKey = self.PublicKey,
+                AgentVersion = self.AgentVersion
+            };
+            MultiAddress addr = "/ip4/0.0.0.0/tcp/0";
+            var swarmA = new SwarmService {LocalPeer = peerA};
+            var peerB = new Peer
+            {
+                Id = other.Id,
+                PublicKey = other.PublicKey,
+                AgentVersion = other.AgentVersion
+            };
+            var swarmB = new SwarmService {LocalPeer = peerB};
+            await swarmA.StartAsync();
+            await swarmB.StartAsync();
+            try
+            {
+                var another = await swarmA.StartListeningAsync(addr);
+                Assert.IsFalse(peerA.Addresses.Contains(addr));
+                Assert.IsTrue(peerA.Addresses.Contains(another));
+
+                await swarmB.ConnectAsync(another);
+                Assert.IsTrue(swarmB.KnownPeers.Contains(peerA));
+
+                // TODO: Assert.IsTrue(swarmA.KnownPeers.Contains(peerB));
+
+                await swarmA.StopListeningAsync(addr);
+                Assert.AreEqual(0, peerA.Addresses.Count());
+            }
+            finally
+            {
+                await swarmA.StopAsync();
+                await swarmB.StopAsync();
+            }
+        }
+
+        [TestMethod]
+        [TestCategory("IPv6")]
+        public async Task Listening_IPv6Any()
+        {
+            var peerA = new Peer
+            {
+                Id = self.Id,
+                PublicKey = self.PublicKey,
+                AgentVersion = self.AgentVersion
+            };
+            MultiAddress addr = "/ip6/::/tcp/0";
+            var swarmA = new SwarmService {LocalPeer = peerA};
+            var peerB = new Peer
+            {
+                Id = other.Id,
+                PublicKey = other.PublicKey,
+                AgentVersion = other.AgentVersion
+            };
+            var swarmB = new SwarmService {LocalPeer = peerB};
+            await swarmA.StartAsync();
+            await swarmB.StartAsync();
+            try
+            {
+                var another = await swarmA.StartListeningAsync(addr);
+                Assert.IsFalse(peerA.Addresses.Contains(addr));
+                Assert.IsTrue(peerA.Addresses.Contains(another));
+
+                await swarmB.ConnectAsync(another);
+                Assert.IsTrue(swarmB.KnownPeers.Contains(peerA));
+
+                // TODO: Assert.IsTrue(swarmA.KnownPeers.Contains(peerB));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)